src/query/functions-v2/src/scalars/string.rs

```diff
@@ -135,11 +136,12 @@ pub fn register(registry: &mut FunctionRegistry) {
         },
     );
 
-    registry.register_4_arg::<StringType, NumberType<i64>, NumberType<i64>, StringType, StringType, _, _>(
+    registry.register_passthrough_nullable_4_arg::<StringType, NumberType<i64>, NumberType<i64>, StringType, StringType, _, _>(
```

Comment:
```suggestion
registry.register_passthrough_nullable_4_arg::<StringType, NumberType<u64>, NumberType<u64>, StringType, StringType, _, _>(
        "insert",
        FunctionProperty::default(),
        |_, _, _, _| None,
        vectorize_with_builder_4_arg::<StringType, NumberType<u64>, NumberType<u64>, StringType, StringType>(
            |srcstr, pos, len, substr, output| {
            let pos = pos as usize;
            let len = len as usize;
            if pos < 1 || pos > srcstr.len() {
                output.put_slice(srcstr);
            } else {
                let pos = pos - 1;
                output.put_slice(&srcstr[0..pos]);
                output.put_slice(substr);
                if pos + len < srcstr.len() {
                    output.put_slice(&srcstr[(pos + len)..]);
                }
            }
            output.commit_row();
            Ok(())
        }),
    );
```

Reply:
registry.register_passthrough_nullable_4_arg::<StringType, NumberType<i64>, NumberType<i64>, StringType, StringType, _, _>

I think the i64 is right.

insert support like this: insert('Quadratic', -1, 4, 'What')

Reply:
Ok. I was just reading document of snowflake that states: 

> [Valid values are between 1 and one more than the length of the string (inclusive).](https://docs.snowflake.com/en/sql-reference/functions/insert.html#:~:text=Valid%20values%20are%20between%201%20and%20one%20more%20than%20the%20length%20of%20the%20string%20(inclusive))

Reply:
I'm wondering why not make negative number counting from the end just like substring 🤔

Reply:
```sql
root@mysqldb 15:26:28 [(none)]> select version();
+-----------+
| version() |
+-----------+
| 8.0.29    |
+-----------+
1 row in set (0.00 sec)

root@mysqldb 15:26:33 [(none)]> select  insert('Quadratic', -1, 4, 'What');
+------------------------------------+
| insert('Quadratic', -1, 4, 'What') |
+------------------------------------+
| Quadratic                          |
+------------------------------------+
1 row in set (0.00 sec)

```

Maybe it was for mysql compatibility